server/plugin.go

```diff
@@ -146,6 +156,33 @@ func (p *Plugin) OnPluginClusterEvent(c *plugin.Context, ev model.PluginClusterE
 	p.HandleClusterEvent(ev)
 }
 
+func (p *Plugin) MessageWillBePosted(c *plugin.Context, post *model.Post) (*model.Post, string) {
+	// If not enabled in config, ignore.
+	if !p.getConfiguration().EnableCodePreview {
+		return nil, ""
+	}
+
+	if post.UserId == "" {
+		return nil, ""
+	}
+
+	msg := post.Message
+	info, err := p.getGitlabUserInfoByMattermostID(post.UserId)
+	if err != nil {
+		p.API.LogError("error in getting user info", "error", err.Message)
+		return nil, ""
+	}
+	// TODO: make this part of the Plugin struct and reuse it.
```

Comment:
To be honest, I don't find any appropriate reason for doing this. In my opinion, this function should be part of GitlabClient instead of Plugin. What do you think?